### ISSUE TITLE
Tag manager: row actions dead beyond the first 100 entries

### ISSUE THREAD

**Issue by goonerish** [Jahia/tags#71]
## Problem

In jContent → Additional → Tag manager, the per-row actions (**view usages**, **rename**, **delete**) do nothing from the 101st tag entry onward. No dialog opens, no request is sent, no error is shown.

The boundary follows the **entry number, not the page number**:

- at 100 rows per page, page 2 onward is dead
- at 10 rows per page, page 11 onward is dead
- entries 1-100 keep working at any page size

### Steps to reproduce

1. Apply more than 100 tags to a site (e.g. 500 on the home page).
2. Open jContent → Additional → Tag manager.
3. Click **view usages**, **rename** or **delete** on any entry past the 100th. Nothing happens.

Two further symptoms that confirm the cause:

- Search for a tag beyond the first 100. The row appears on page 1, and its buttons are still dead.
- Sort the tag column descending. The dead rows move to page 1; the working set stays the same 100 entries.

### Environment

- `tags` module 8.2.0
- Jahia 8.1 and 8.2

## Cause

The click listeners are bound **once, after DataTables has already removed the off-page rows from the document**.

`dataTablesSettings.init('tableTagsList', 100, ...)` runs synchronously and performs the first draw, which detaches every row except the current page. The next lines then bind listeners by walking the document:

```js
// src/main/resources/jnt_tagsManager/html/tagsManager.settingsBootstrap3GoogleMaterialStyle.flow/tagsManager.jsp:81-85
dataTablesSettings.init('tableTagsList', 100, null, null, null, dtOptions);  // draws page 1, detaches rows 101+
...
attachRenameAndDeleteListeners();   // too late
attachViewUsagesListeners();        // too late
```

```js
// src/main/resources/javascript/tagsManager.js:106
var renameButtons = document.getElementsByClassName("renameTagButton");
for (var i = 0; i < renameButtons.length; i++) {
    renameButtons[i].addEventListener("click", ...);
}
```

`document.getElementsByClassName()` returns only elements **inside the document**, so it finds 100 buttons rather than 500. Rows 101+ never receive a listener, and nothing re-binds them: the attach functions run once per page load.

The `100` is the `iDisplayLength` passed on the line above. That is why the boundary is the 101st entry.

Entries 1-100 keep working at any page size because DataTables caches each row's `<tr>` node and re-appends the same node on every redraw (`body.children().detach(); body.append($(anRows));`). Detaching a node does not remove its listeners. Rows 101+ are re-appended just as faithfully, with the zero listeners they were created with.

**Comment by goonerish** [comment]
## Suggested fix: event delegation

Bind one delegated listener per action to the table element instead of one listener per button. DataTables never replaces the table itself — it only detaches and re-appends the `<tbody>` children — so a delegated handler matches rows on every page, and the init-vs-attach ordering stops mattering.

### `src/main/resources/javascript/tagsManager.js` — table `tableTagsList`

Replaces the `getElementsByClassName` loops at `:105-128`.

```js
function attachRenameAndDeleteListeners() {
    $("#tableTagsList").on("click", ".renameTagButton", function() {
        bbRenameTag(this.id.replace("rename_", ""));
    });

    $("#tableTagsList").on("click", ".deleteTagButton", function() {
        bbDeleteTag(this.id.replace("delete_", ""));
    });
}

function attachViewUsagesListeners() {
    $("#tableTagsList").on("click", ".viewUsageButton", function() {
        viewUsages(this.id.replace("usage_", ""));
    });
}
```

### `src/main/resources/javascript/tagUsages.js` — table `tableTagDetails`

Replaces the loops at `:89-103`.

```js
function attachRenameAndDeleteListeners() {
    $("#tableTagDetails").on("click", ".renameTagButton", function() {
        bbRenameTag(this.id.replace("rename_", ""));
    });

    $("#tableTagDetails").on("click", ".deleteTagButton", function() {
        bbDeleteTag(this.id.replace("delete_", ""));
    });
}
```
